src/axom/quest/DistributedClosestPoint.hpp

```diff
@@ -408,11 +411,88 @@ class DistributedClosestPointImpl
     return false;
   }
 
+  /// Get local copy of all ranks BVH root bounding boxes.
+  void gatherBVHRoots()
+  {
+    SLIC_ASSERT_MSG(
+      isBVHTreeInitialized(),
+      "BVH tree must be initialized before calling 'gatherBVHRoots");
+
+    BoxType local_bb;
+    switch(m_runtimePolicy)
+    {
+    case RuntimePolicy::seq:
+      local_bb = m_bvh_seq->getBounds();
+      break;
+
+    case RuntimePolicy::omp:
+#ifdef _AXOM_DCP_USE_OPENMP
+      local_bb = m_bvh_omp->getBounds();
+#else
+      break;
+#endif
+
+    case RuntimePolicy::cuda:
+#ifdef _AXOM_DCP_USE_CUDA
+      local_bb = m_bvh_cuda->getBounds();
+#else
+      break;
+#endif
+    }
+
+    gatherBoundingBoxes(local_bb, m_objectPartitionBbs);
+  }
+
+  /// Allgather one bounding box from each rank.
+  void gatherBoundingBoxes(const BoxType& aabb, BoxArray& all_aabbs) const
+  {
+    // Using MPI calls.  Should we change to Conduit relay MPI?
+    Array<double> sendbuf(2 * DIM);
+    aabb.getMin().to_array(&sendbuf[0]);
+    aabb.getMax().to_array(&sendbuf[DIM]);
+    Array<double> recvbuf(m_nranks * sendbuf.size());
+    int errf = MPI_Allgather(sendbuf.data(),
+                             2 * DIM,
+                             mpi_traits<double>::type,
+                             recvbuf.data(),
+                             2 * DIM,
+                             mpi_traits<double>::type,
+                             MPI_COMM_WORLD);
+    SLIC_ASSERT(errf == MPI_SUCCESS);
+
+    all_aabbs.clear();
+    all_aabbs.reserve(m_nranks);
+    for(int i = 0; i < m_nranks; ++i)
+    {
+      PointType lower(&recvbuf[i * 2 * DIM]);
+      PointType upper(&recvbuf[i * 2 * DIM + DIM]);
+      all_aabbs.emplace_back(BoxType(lower, upper));
+    }
+  }
+
+  /// Compute bounding box for local part of a mesh.
+  BoxType computeMeshBoundingBox(conduit::Node& mesh,
+                                 const std::string& coordset) const
+  {
+    auto& coords = mesh[fmt::format("coordsets/{}/values", coordset)];
+    SLIC_ASSERT(internal::extractDimension(coords) == NDIMS);
+    const int npts = internal::extractSize(coords);
+    BoxType rval;
+    ArrayView<PointType> queryPts = ArrayView_from_Node<PointType>(
+      mesh.fetch_existing("coordsets/coords/values/x"),
+      npts);
+    for(const auto& p : queryPts)
+    {
+      rval.addPoint(p);
+    }
+    return rval;
+  }
```

Comment:
Let's talk about it.  Much of the CPU/GPU automation still seems like magic to me.

Reply:
@gunney1 @kennyweiss walking trough the CPU/GPU implementation mechanics in Axom would be a good topic for a team meeting in the near future, I think. Not everyone on the team has worked on GPU porting and it would be good to have everyone on the same page. Agree?

Reply:
I think that would be a really good idea!